chainstate/tx-verifier/src/transaction_verifier/input_output_policy/constraints_accumulator.rs

```diff
@@ -13,60 +13,94 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::collections::BTreeMap;
+use std::collections::{btree_map::Entry, BTreeMap};
 
 use common::{
     chain::{
-        timelock::OutputTimeLock, AccountCommand, AccountSpending, ChainConfig, PoolId, TxInput,
-        TxOutput,
+        output_value::OutputValue,
+        timelock::OutputTimeLock,
+        tokens::{TokenData, TokenId},
+        AccountCommand, AccountSpending, ChainConfig, DelegationId, PoolId, Transaction, TxInput,
+        TxOutput, UtxoOutPoint,
     },
-    primitives::{Amount, BlockDistance, BlockHeight},
+    primitives::{Amount, BlockHeight, Id},
 };
 use utils::ensure;
 
+use crate::{transaction_verifier::amounts_map::CoinOrTokenId, Fee};
+
 use super::IOPolicyError;
 
+type BlockCount = u64;
```

Comment:
Isn't it better to use NonZeroU64?

Reply:
right, you mentioned this before and I tried it out. Maturity distance can be 0 and it actually is during IgnoreConsensus or PoW. If I make it NonZero I have to handle this specific case separately and account such coins in `unconstrained_values` instead which imho make the code more complicated.